internal/services/db/transfer.go

```diff
@@ -263,6 +263,21 @@ func (db *TransferDB) TransferExists(txHash string) (bool, error) {
 	return count > 0, nil
 }
 
+// TransferSimilarExists returns true if the transfer tx_hash is empty and to, from and value match in the db
+func (db *TransferDB) TransferSimilarExists(from, to, value string) (bool, error) {
+	var count int
+	row := db.rdb.QueryRow(fmt.Sprintf(`
+	SELECT COUNT(*) FROM t_transfers_%s WHERE tx_hash = '' AND from_addr = $1 AND to_addr = $2 AND value = $3
```

Comment:
What about adding a condition on the timestamp? No need to scan the entire DB.

Reply:
There is no definite timestamp until a transaction gets added on chain and it's only in the case where we send something that there is every anything with `tx_hash = ''`. 

So by eliminating everything else, it's already just the transactions that need reconciling . But we are also specific with the `from` and `to` and `value`. So, it's a very specific query.

And there are indexes.

So, given this query on USDC (which now has 25,270,116 indexed transactions):
```
EXPLAIN ANALYZE SELECT hash, tx_hash, token_id, created_at, from_to_addr, from_addr, to_addr, nonce, value, data, status
		FROM t_transfers_137_0x2791bca1f2de4661ed88a30c99a7a9449aa84174
		WHERE tx_hash = '' AND from_addr = '0x75ba54cfa1f6e6b4dece74c6020d6dbda6524732' AND to_addr = '0xDA9e31F176F5B499b3B208a20Fe169b3aA01BA26' AND value = '1000';
```

This is the result. Only indexes are used. Execution time is decent.
```
Index Scan using idx_transfers_137_0x2791bcaa9449aa84174_to_addr_from_addr_value on t_transfers_137_0x2791bca1f2de4661ed88a30c99a7a9449aa84174  (cost=0.69..8.71 rows=1 width=338) (actual time=1.965..1.965 rows=0 loops=1)
  Index Cond: ((to_addr = '0xDA9e31F176F5B499b3B208a20Fe169b3aA01BA26'::text) AND (from_addr = '0x75ba54cfa1f6e6b4dece74c6020d6dbda6524732'::text) AND (value = '1000'::text))
  Filter: (tx_hash = ''::text)
Planning Time: 18.625 ms
Execution Time: 2.022 ms
```